app/models/grda_warehouse/version.rb

```diff
@@ -42,5 +60,45 @@ def whodunnit_impersonator_pattern
       # When impersonating a user, whodunnit is recorded as "<true_user> as <current_user>"
       /^(\d+) as (\d+)$/
     end
+
+    protected
+
+    # Extracted this method from the controller. It appears to be a fallback for old versions didn't include changes
```

Comment:
If we need this, I think it would be on the user audit screens or audit reports.